### ISSUE TITLE
webpack externals for sw.js compilation

### ISSUE THREAD

**Issue by yorrd** [GoogleChrome/workbox#2724]
**Library Affected**:
*workbox-webpack-plugin*

**Issue or Feature Request Description**:
In our code base, we use [webpack externals](https://webpack.js.org/configuration/externals/#externals) quite regularly. I'm trying to set up a service worker using workbox and noticed that I can't influence the webpack config used to build the service worker file apart from adding plugins. The externals option is a direct child of the options object. 

The reason I want such a complex configuration for the service worker file is that I'm looking into building a database connector that works offline "out of the box" on our stack and therefore needs access to the more complex libraries we use in the rest of the code base. 

**tl;dr**: I need the externals option from webpack to be exposed for the `sw.js` compilation.

Is this something that you would accept a pull request for? Am I understanding something wrong or missing a better way to do this?

**Comment by jeffposnick** [comment]
If you're using the `InjectManifest` plugin (which I think you are), it will trigger a [child compilation](https://webpack.js.org/api/compilation-object/#createchildcompiler), inheriting settings from the parent compilation. (We also support allowing you to add in some specific plugins just to the child compilation, as you noted.) While I don't personally have experience using externals with `webpack`, I am a little surprised that the child compilation doesn't just inherit your externals configuration from the main compilation. At the same time, `webpack` child compilations are [notoriously](https://medium.com/@prateekbh/my-experience-writing-a-webpacks-child-compiler-plugin-a1237c175947)... difficult to configure properly.

Taking a step back, though, the only way you could import external code in a service worker is using `importScripts()`. I don't see anything in the `webpack` externals documentation indicating that its "understands' `importScripts()`, so I wonder if using that feature would give you the behavior you're looking for anyway.

What if you just hardcode a call to `importScripts('path/to/dependency.js')` in the file that you use as your `swSrc`. Is that feasible for your use case?

**Comment by yorrd** [comment]
@jeffposnick thanks for your super helpful response. That turned out to be a great workaround. 

Actually, the problem wasn't even that the externals aren't copied to the child compilation (they are), it's just that I need different externals than for the main application. The `importScript` now imports a `.js` file that is emitted by a different webpack configuration altogether which just runs in the background in watch mode. 

Thanks in particular for the link on webpack child compilations, that cleared up several misunderstandings I had about child compilations. 

Consider this closed, `importScripts()` is just what I wanted :)
